backend/api/Services/AccessRoleService.cs

```diff
@@ -7,25 +7,37 @@ namespace Api.Services
     public interface IAccessRoleService
     {
         public Task<List<string>> GetAllowedInstallationCodes();
+        public Task<List<string>> GetAllowedInstallationCodes(List<string> roles);
     }
 
-    public class AccessRoleService(FlotillaDbContext context, IHttpContextAccessor httpContextAccessor, IInstallationService installationService) : IAccessRoleService
+    public class AccessRoleService(FlotillaDbContext context, IHttpContextAccessor httpContextAccessor) : IAccessRoleService
     {
         private const string SUPER_ADMIN_ROLE_NAME = "Role.Admin";
 
         public async Task<List<string>> GetAllowedInstallationCodes()
+        {
+            if (httpContextAccessor.HttpContext == null)
+            {
+                return await context.Installations.Select((i) => i.InstallationCode.ToUpperInvariant()).ToListAsync();
+            }
+
+            var roles = httpContextAccessor.HttpContext.GetRequestedRoleNames();
+
+            return await GetAllowedInstallationCodes(roles);
+        }
+
+        public async Task<List<string>> GetAllowedInstallationCodes(List<string> roles)
         {
             var dbRoles = await context.AccessRoles.Include((r) => r.Installation).ToListAsync();
-            var roles = httpContextAccessor.HttpContext!.GetRequestedRoleNames();
 
             if (roles.Contains(SUPER_ADMIN_ROLE_NAME))
             {
-                return (await installationService.ReadAll()).Select((i) => i.InstallationCode).ToList();
+                return await context.Installations.Select((i) => i.InstallationCode).ToListAsync();
             }
             else
             {
                 return await context.AccessRoles.Include((r) => r.Installation)
-                    .Where((r) => roles.Contains(r.RoleName)).Select((r) => r.Installation != null ? r.Installation.InstallationCode : "").ToListAsync();
+                    .Where((r) => roles.Contains(r.RoleName)).Select((r) => r.Installation != null ? r.Installation.InstallationCode.ToUpperInvariant() : "").ToListAsync();
```

Comment:
Why invariant?

Reply:
Invariant is done consistently regardless of culture. Since this is in an entity framework call it is being converted to SQL, which means that you are not allowed to specify culture. But on the other hand, omitting culture when using ToUpper will cause a linting error, unless we disable the linting for this line. So I though this was simplest. It will behave identically to ToUpper anyways